blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1001,6 +1002,51 @@ fn sync_tipset<DB: BlockStore + Sync + Send + 'static, C: Consensus>(
     })
 }
 
+async fn fetch_batch<DB: BlockStore + Send + Sync + 'static, C: Consensus>(
+    batch: &[Arc<Tipset>],
+    network: &SyncNetworkContext<DB>,
+    chainstore: &ChainStore<DB>,
+    s: &channel::Sender<FullTipset>,
+) -> Result<(), TipsetRangeSyncerError<C>> {
+    // Tipsets in `batch` are already in chronological order
+    if let Some(head) = batch.last() {
+        let epoch = head.epoch();
+        let len = batch.len();
+
+        debug!("ChainExchange message sync tipsets: epoch: {epoch}, len: {len}");
+
+        let compacted_messages = network
+            .chain_exchange_messages(None, head.key(), len as u64)
+            .await
+            .map_err(TipsetRangeSyncerError::NetworkMessageQueryFailed)?;
+        let messages_iter = compacted_messages.into_iter().rev();
+
+        // Since the bundle only has messages, we have to put the headers in them
+        for (messages, tipset) in messages_iter.zip(batch.iter()) {
```

Comment:
One-lining this would make it a bit easier to read for me. Not a big deal though.

Reply:
Done.